File: task5.cpp
#include <iostream>
using namespace std;

void main() {
    int number, degree1, degree2;
    
    cout << "Input number: ";
    cin >> number;
    cout << "Input degree: ";
    cin >> degree1;
    cout << "Input degree for degree: ";
    cin >> degree2;

    int numberInDegree = 1;                            // результат возведения в первую степень
    for (int i = 1; i <= degree1; i++) {               // возводим число в первую степень
        numberInDegree = (numberInDegree * number) % 10; // смотрим последнюю цифру 
    }

    int numberInDegree2 = 1;                            // результат возведения во вторую степень
    for (int i = 1; i <= degree2; i++) {                // возводим число в степени во вторую степень
        numberInDegree2 = (numberInDegree2 * numberInDegree) % 10; // смотрим последнюю цифру
    }

    cout << "Последняя цифра "<< numberInDegree2 << endl;
}

File: Go/task5.go
package main

import "fmt"

func main() {
	var number, degree1, degree2 int

	fmt.Println("Input number: ")
	fmt.Scanln(&number)
	fmt.Println("Input degree: ")
	fmt.Scanln(&degree1)
	fmt.Println("Input degree for degree: ")
	fmt.Scanln(&degree2)

	numberInDegree1 := 1            // результат возведения в первую степень
	for i := 1; i <= degree1; i++ { // возводим число в первую степень
		numberInDegree1 = (numberInDegree1 * number) % 10 // смотрим последнюю цифру
	}

	numberInDegree2 := 1            // результат возведения во вторую степень
	for i := 1; i <= degree2; i++ { // возводим число в степени во вторую степень
		numberInDegree2 = (numberInDegree2 * numberInDegree1) % 10 // смотрим последнюю цифру
	}

	fmt.Println("The last digit in the number is in the degree:", numberInDegree2)
}
